File: src/6/6.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false
});

let cols = []
let lines = []
console.log("Go!")
rl.on('line', (line) => {
    lines.push(line)
    line.split(/\s+/).forEach((el, i) => {
        if (!cols[i]) {
            cols[i] = []
        }
        if (el != "+" && el != "*") {
            cols[i].push(el);
        } else {
            cols[i].unshift(el);
        }
    })
})


rl.on('close', () => {
    let sum1 = cols.map((col, _, arr) => {
        if (col[0] == '+') {
            let sum = 0;
            col.slice(1).forEach((a) => sum += +a);
            return sum;
        } else {
            let sum = 1;
            col.slice(1).forEach((a) => sum *= +a);
            return sum;
        }
    }).reduce((a, b) => a + b);
    console.log(sum1);
    let sum2 = 0;
    let buf = 0;
    let op = '';
    for (let i = 0; i < lines[0].length; i++) {
        let br = true;
        let number = ''
        for (let j = 0; j < lines.length - 1; j++) {
            if (lines[j][i] !== ' ') {
                br = false
                number += lines[j][i]
            }
        }
        if (lines[lines.length - 1][i] !== ' ') {
            op = lines[lines.length - 1][i];
        }
        if (br && i !== lines[0].length - 1) {
            sum2 += buf
        } else {
            if (op == '+') {
                buf += +number
            } else if (op == "*") {
                if (buf == 0) {
                    buf = 1
                }
                buf *= +number
            }
        }
        if (i == lines[0].length - 1) {
            sum2 += buf
        }
        if (br) {
            buf = 0
        }
    }
    console.log(sum2)
})

File: src/6/6.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
)

func main() {
	input := bufio.NewScanner(os.Stdin)
	lines := make([]string, 5)
	cols := make([][]string, 750)
	exp := regexp.MustCompile(`\s+`)
	fmt.Println("Go!")
	for input.Scan() {
		line := input.Text()
		lines = append(lines, line)
		split := exp.Split(line, -1)
		for i, val := range split {
			if i >= len(cols) {
				cols = append(cols, []string{})
			}
			if val == "*" || val == "+" {
				cols[i] = append([]string{val}, cols[i]...)
			} else {
				cols[i] = append(cols[i], val)
			}
		}
	}
	var sum1 int64 = 0
	for _, arr := range cols {
		op := arr[0]
		var buf int64
		if op == "+" {
			buf = 0
		} else {
			buf = 1
		}
		for _, val := range arr[1:] {
			if op == "+" {
				val, _ := strconv.ParseInt(val, 0, 64)
				buf += val
			} else {
				val, _ := strconv.ParseInt(val, 0, 64)
				buf *= val
			}
		}
		sum1 += buf
	}
	var sum2 int64 = 0
	var buf int64 = 0
	op := ""
	for i := range lines[0] {
		br := true
		number := ""
		for j := 0; j < len(lines)-1; j++ {
			if lines[j][i] != ' ' {
				br = false
				number += string([]rune(lines[j])[i])
			}
		}
		if lines[len(lines)-1][i] != ' ' {
			op = string(lines[len(lines)-1][i])
		}
		if br && i != len(lines[0])-1 {
			sum2 += buf
		} else {
			switch op {
			case "+":
				num, _ := strconv.Atoi(number)
				buf += int64(num)
			case "*":
				if buf == 0 {
					buf = 1
				}
				num, _ := strconv.Atoi(number)
				buf *= int64(num)
			}
		}
		if i == len(lines[0])-1 {
			sum2 += buf
		}
		if br {
			buf = 0
		}
	}
	fmt.Println(sum1, sum2)
}
